### ISSUE TITLE
File Chunks

### ISSUE THREAD

**Issue by MoaazSuliman** [paulcwarren/spring-content#2074]
Hi All,
I'm uploading the large file to azure,
Is there are chunk why to can use it for large files?!

**Comment by paulcwarren** [comment]
Hi @MoaazSuliman ,

What is your issue?  Are you encountering OOMs?  Or is it just slow?  If it is the latter then we could add a property to allow application to specify the buffer size to use like we did recently on JPA when we fixed issue #1944

**Comment by MoaazSuliman** [comment]
@paulcwarren 
No I want to divide my file into chunks to can upload the large files without any problems for connection..
is Spring content support chunks?!

**Comment by paulcwarren** [comment]
>>> do you support file chunks while uploading
No, Spring Content does not support that currently.  Supporting byte-ranges on uploads would be the other way we could consider supporting that recovery use case.

**Comment by MoaazSuliman** [comment]
@paulcwarren 
is there are documentation for byte-ranges , to know how can I use it.
and If the failure happen , the uploading will restart from the beginning?!

**Comment by paulcwarren** [comment]
SC does not support either currently (byte-ranges are supported on GETs only atm).  But byte range PUTs would be less disruptive to the API.   Are you using Spring Content REST, or just one of the storage modules?
